src/main/java/com/puppycrawl/tools/checkstyle/checks/design/VisibilityModifierCheck.java

```diff
@@ -204,31 +351,40 @@
  * <pre>
  * &lt;module name=&quot;VisibilityModifier&quot;&gt;
  *   &lt;property name=&quot;allowPublicImmutableFields&quot; value=&quot;true&quot;/&gt;
- *   &lt;property name=&quot;immutableClassCanonicalNames&quot; value=&quot;
- *   com.google.common.collect.ImmutableSet&quot;/&gt;
+ *   &lt;property name=&quot;immutableClassCanonicalNames&quot;
+ *   value=&quot;com.google.common.collect.ImmutableSet&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
  * <p>
  * Example of allowed public immutable fields:
  * </p>
  * <pre>
- * public class ImmutableClass
- * {
- *   public final ImmutableSet&lt;String&gt; includes; // No warning
- *   public final ImmutableSet&lt;String&gt; excludes; // No warning
- *   public final java.lang.String notes; // Warning here because
- *                                        //'java.lang.String' wasn't specified as allowed class
- *   public final int someValue; // No warning
- *
- *   public ImmutableClass(Collection&lt;String&gt; includes, Collection&lt;String&gt; excludes,
- *                String notes, int someValue)
- *   {
- *     this.includes = ImmutableSet.copyOf(includes);
- *     this.excludes = ImmutableSet.copyOf(excludes);
- *     this.value = value;
- *     this.notes = notes;
- *     this.someValue = someValue;
- *   }
+ * public final class MyClass {
+ * private int myPrivateField1;
+ *
+ * int field1;               // violation, must have a visibility modifier
+ *
+ * protected String field2;  // violation, protected visibility is not allowed
+ *
+ * public int field3 = 42;   // violation, not static final, immutable, matching the pattern
+ * public long serialVersionUID = 1L;
+ * public static final int field4 = 42;
+ *
+ * public final int field5 = 42;
+ * public final java.lang.String notes;   // violation, public immutable fields are not allowed
+ *
+ * public final Set&lt;String&gt; mySet1 = new HashSet&lt;&gt;();  // violation, HashSet is mutable
+ * public final ImmutableSet&lt;String&gt; mySet2;           // violation, mutable
```

Comment:
should not be violation, as we defined set in config

Reply:
This is exactly what i told you before and you told me to just show how they work and if there's any issue you will open an issue for it.
Here's the output:
```
[ERROR] E:\GSoC\Test6.java:13:32: Variable 'notes' must be private and have accessor methods. [VisibilityModifier]
[ERROR] E:\GSoC\Test6.java:15:27: Variable 'mySet1' must be private and have accessor methods. [VisibilityModifier]
[ERROR] E:\GSoC\Test6.java:16:36: Variable 'mySet2' must be private and have accessor methods. [VisibilityModifier]
```

Reply:
ok , keep it it works now.

please create issue on this , as we found a bug.

Reply:
it is not a bug, as Check does not know that String is immutable
update config to be:
```
<property name="immutableClassCanonicalNames"
        value="com.google.common.collect.ImmutableSet, jav.lang.String"/>
```

to be more realistic.

Reply:
Greate, i will push the updates now

Reply:
> it is not a bug, as Check does not know that String is immutable
update config

Even after updating it `notes` and `mySet1` still violates.

Reply:
> Even after updating it still have mySet1 as a volitation.

Since `Hashset` is mutable, so now it does make sense and only `mySet1` will violate right?

Reply:
Something is strange, please do last update, in worse case we will create issue on this to fix later.

Let's merge this PR to just show how it works now.

Reply:
I have another question, if config is like this:
```
  <property name="ignoreAnnotationCanonicalNames"
  value="CustomAnnotation"/>
```
then something like 
```
 @com.annotation.CustomAnnotation
  String customAnnotated1; 
```
shouldn't produce a violation, right?

Reply:
Property needs full name of class, you provided short name. It might be a reason of problem